services/pama-imaging.js

```diff
@@ -10,58 +12,78 @@ const router = express.Router();
  * @param {*} entries a list of resources.
  * @param {*} selections a list of selected resource strings matching the form 'Type/Id'.
  */
-function* findResources(entries, selections) {
-  for (let i = 0; i < selections.length; i += 1) {
-    const [resourceType, resourceId] = selections[i].split('/');
-    const resources = entries
-      .map(e => e.resource)
-      .filter(r => r.resourceType === resourceType && r.id === resourceId);
-    for (const resource of resources) {
-      yield resource;
-    }
-  }
+const findResources = (entries, selections) =>
+  selections
+    .map(s => s.split('/'))
+    .map(([resourceType, resourceId]) =>
+      entries
+        .map(e => e.resource)
+        .filter(r => r.resourceType === resourceType && r.id === resourceId)[0]);
+
+
+const SNOMED = 'http://snomed.info/sct'
+const CPT = 'http://www.ama-assn.org/go/cpt'
+
+// TODO: populate this with more data.
+const appropriateCPTReasons = {
+  '72133': { SNOMED: [['279039007']].map(x => new Set(x)) },
+};
+
+function findCodes(codes, systemName) {
+  return codes.filter(c => c.coding.system === systemName).map(c => c.code);
```

Comment:
If I'm understanding right: `codes` is an array of `CodeableConcept`s, each of which looks like http://build.fhir.org/datatypes#CodeableConcept.

Now, each entry is a `CodeableConcept, so `c` in the anonymous functions above is a `CodeableConcept`. That means `c.coding` is an *array*, so it won't ever have a `.system` property. So I'm thinking `findCodes` only ever returns an array. Am I off base?